src/cargo/core/compiler/standard_lib.rs

```diff
@@ -99,8 +102,13 @@ pub fn resolve_std<'cfg>(
         /*dev_deps*/ false, &features, /*all_features*/ false,
         /*uses_default_features*/ true,
     );
-    let resolve = ops::resolve_ws_with_opts(&std_ws, opts, &specs)?;
-    Ok((resolve.pkg_set, resolve.targeted_resolve))
+    let resolve =
+        ops::resolve_ws_with_opts(&std_ws, target_data, requested_target, &opts, &specs, false)?;
```

Comment:
Reading up on this, the `false` here seems pretty special, so could it perhaps transition to an enum with named stuff? It's definitely more verbose, but we probably want to err on the side of making sure this is as resilient as can be!